strings/manacher.nim

```diff
@@ -0,0 +1,88 @@
+# Manacher's algorithm
+#[
+Determine the longest palindrome in a string in linear time in its length
+with Manacher's algorithm
+Inspired from:
+https://github.com/jilljenn/tryalgo/blob/master/tryalgo/manacher.py
+https://en.wikipedia.org/wiki/Longest_palindromic_substring
+]#
+import std/[strutils, sequtils, sets]
+{.push raises: [].}
+
+runnableExamples:
+  let example1 = "cabbbab"
+  doAssert(manacherString(example1) == "abbba")
+  doAssert(manacherIndex(example1) == 1 .. 5)
+  doAssert(manacherLength(example1) == 5)
+
+func manacherIndex*(s: string): HSlice[int, int] {.raises: [ValueError].} =
+    #[Longest palindrome in a string by Manacher
+      :param s: string, lowercase ascii, no whitespace
+      :returns: indexes i,j such that s[i:j] is the longest palindrome in s
+      :complexity: O(len(s))
+      All the indexes refer to an intermediate string t
+      of the form "^#a#b#a#a#$" for s="abaa"
+    ]#
+    if s.len == 0:
+      raise newException(ValueError, "Empty string")
+    let extraSymbols = toHashSet(['$', '^', '#'])
```

Comment:
When I compiled the code: `prog.nim(28, 24) Error: undeclared identifier: 'toHashSet'` ...

Reply:
I do not have this error. What OS and which Nim version do you use?

Reply:
Should be resolved.